core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -514,24 +514,28 @@ public void onFailure(Exception e) {
                             request),
                     e);
                 request.onRetry();
-                final ThreadContext.StoredContext context = threadPool.getThreadContext().newStoredContext();
+                final Supplier<ThreadContext.StoredContext> context = threadPool.getThreadContext().newRestorableContext(false);
```

Comment:
Is this needed? I think the observer takes care of this? If not, I can work on folding this into the observer as a follow up.